modules/freewheelSSPBidAdapter.js

```diff
@@ -0,0 +1,303 @@
+  * @return ServerRequest Info describing the request to the server.
+  */
+  buildRequests: function(bidRequests) {
+    // var currency = config.getConfig(currency);
+
+    this._currentBidRequest = bidRequests[0];
+    if (bidRequests.length > 1) {
+      console.log('Prebid.JS - freewheel bid adapter: only one ad unit is required.');
+    }
+
+    var requestParams = {
+      reqType: 'AdsSetup',
+      protocolVersion: '2.0',
+      zoneId: this._currentBidRequest.params.zoneId,
+      componentId: getComponentId(this._currentBidRequest.params.format)
+    };
+
+    var location = utils.getTopWindowUrl();
+    if (isValidUrl(location)) {
+      requestParams.loc = location;
+    }
+
+    this._currentPlayerSize = getBiggerSize(this._currentBidRequest.sizes);
+    if (this._currentPlayerSize[0] > 0 || this._currentPlayerSize[1] > 0) {
+      requestParams.playerSize = this._currentPlayerSize[0] + 'x' + this._currentPlayerSize[1];
+    }
+
+    return {
+      method: 'GET',
+      url: FREEWHEEL_ADSSETUP,
+      data: requestParams
+    };
+  },
+
+  /**
+  * Unpack the response from the server into a list of bids.
+  *
+  * @param {*} serverResponse A successful response from the server.
+  * @return {Bid[]} An array of bids which were nested inside the server.
+  */
+  interpretResponse: function(serverResponse) {
+    if (typeof serverResponse == 'object' && typeof serverResponse.body == 'string') {
+      serverResponse = serverResponse.body;
+    }
+
+    var xmlDoc;
+    try {
+      var parser = new DOMParser();
+      xmlDoc = parser.parseFromString(serverResponse, 'application/xml');
+    } catch (err) {
+      console.warn('Prebid.js - ' + BIDDER_CODE + ' : ' + err);
+      return;
+    }
+
+    const princingData = getPricing(xmlDoc);
+    const creativeId = getCreativeId(xmlDoc);
+
+    const topWin = getTopMostWindow();
+    if (!topWin.freeheelssp_cache) {
+      topWin.freeheelssp_cache = {};
+    }
+    topWin.freeheelssp_cache[this._currentBidRequest.adUnitCode] = serverResponse;
+
+    const bidResponses = [];
+
+    if (princingData.price) {
+      const bidResponse = {
+        requestId: this._currentBidRequest.bidId,
+        bidderCode: this._currentBidRequest.bidder,
+        cpm: princingData.price,
+        width: this._currentPlayerSize[0],
+        height: this._currentPlayerSize[1],
+        creativeId: creativeId,
+        currency: princingData.currency,
+        netRevenue: true,
+        ttl: 360
+      };
+
+      var mediaTypes = this._currentBidRequest.mediaTypes || {};
+      if (mediaTypes.video) {
+        bidResponse.vastXml = serverResponse;
+        bidResponse.mediaType = 'video';
+      } else {
+        bidResponse.ad = formatAdHTML(this._currentBidRequest, this._currentPlayerSize)
+      }
+
+      bidResponses.push(bidResponse);
+    }
+
+    return bidResponses;
+  },
+
+  getUserSyncs: function(syncOptions) {
+    var img = new Image();
+    img.src = USER_SYNC_URL;
```

Comment:
You need to return the sync and let Prebid.js attach it to the page.  It is not allowed to add your own syncs in 1.0.  You can look at an example of how to return a properly formatted sync in one of the other 1.0 adapters like rubicon or appnexus.

Reply:
Done thanks.